core/cloudflow-operator/src/main/scala/cloudflow/operator/CloudflowApplication.scala

```diff
@@ -122,30 +122,75 @@ object CloudflowApplication {
   }
 
   object Status {
+    val Unknown          = "Unknown"
+    val Running          = "Running"
+    val Pending          = "Pending"
+    val CrashLoopBackOff = "CrashLoopBackOff"
+
     def apply(
-        app: CloudflowApplication.CR,
-        streamletStatuses: Vector[StreamletStatus] = Vector()
-    ): Status =
+        spec: CloudflowApplication.Spec
+    ): Status = {
+      val streamletStatuses = createStreamletStatuses(spec)
       Status(
-        app.spec.appId,
-        app.spec.appVersion,
-        streamletStatuses
+        spec.appId,
+        spec.appVersion,
+        streamletStatuses,
+        Unknown
       )
+    }
+
+    def createStreamletStatuses(spec: CloudflowApplication.Spec) = {
+      // TODO this code needs to be improved.
+      // There are some plans to make replicas mandatory
+      // and to indicate extraPods required in the deployment to prevent this issue
+      import cloudflow.operator.runner._
```

Comment:
Does this take into account the current scaling factor?

Reply:
It takes into account the replicas which is the scaling factor.